tests/test_pytypes.cpp

```diff
@@ -11,6 +11,34 @@
 
 #include <utility>
 
+namespace external {
+namespace detail {
+bool check(PyObject *o) { return PyFloat_Check(o) != 0; }
+
+PyObject *conv(PyObject *o) {
+    if (PyLong_Check(o)) {
+        double v = PyLong_AsDouble(o);
+        if (v == -1.0 && PyErr_Occurred()) {
+            return nullptr;
+        }
+        return PyFloat_FromDouble(v);
+    } else {
+        PyErr_SetString(PyExc_TypeError, "Unexpected type");
+        return nullptr;
+    }
+}
+
+PyObject *default_constructed() { return PyFloat_FromDouble(0.0); }
+} // namespace detail
+class float_ : public py::object {
+    PYBIND11_OBJECT_CVT(float_, py::object, external::detail::check, external::detail::conv)
+
+    float_() : py::object(external::detail::default_constructed(), borrowed_t{}) {}
```

Comment:
Is this the leak? I'm thinking it needs to be `stolen_t` (to take ownership).

("stolen" always sounds wrong to me btw)

Reply:
Yes, right. Yes in numpy.h `ensure` is using `reinterpret_steal`. Another fix is on its way.